src/utils/subgraph.ts

```diff
@@ -0,0 +1,120 @@
+import { gql, DocumentNode, ApolloQueryResult } from '@apollo/client'
+import { DDO } from '@oceanprotocol/lib'
+import { getApolloClientInstance } from '../providers/ApolloClientProvider'
+import BigNumber from 'bignumber.js'
+
+const freQuery = gql`
+  query AssetFrePrice($datatoken_in: [String!]) {
+    fixedRateExchanges(orderBy: id, where: { datatoken_in: $datatoken_in }) {
+      rate
+      id
+      datatoken {
+        id
+        address
+      }
+    }
+  }
+`
+
+const poolQuery = gql`
+  query AssetPoolPrice($datatokenAddress_in: [String!]) {
+    pools(where: { datatokenAddress_in: $datatokenAddress_in }) {
+      spotPrice
+      id
+      datatokenAddress
+    }
+  }
+`
+
+const previousOrderQuery = gql`
+  query AssetPreviousOrder($id: String!, $account: String!) {
+    tokenOrders(
+      first: 1
+      where: { datatokenId: $id, payer: $account }
+      orderBy: timestamp
+      orderDirection: desc
+    ) {
+      timestamp
+      tx
+    }
+  }
+`
+
+async function fetchData(
+  query: DocumentNode,
+  variables: any
+): Promise<ApolloQueryResult<any>> {
+  try {
+    const client = getApolloClientInstance()
+    const response = await client.query({
+      query: query,
+      variables: variables
+    })
+    return response
+  } catch (error) {
+    console.error('Error fetchData: ', error.message)
+  }
+}
+
+export async function getPreviousOrders(
+  id: string,
+  account: string,
+  assetTimeout: string
+): Promise<string> {
+  const variables = {
+    id: id,
+    account: account
+  }
+  const fetchedPreviousOrders: any = await fetchData(
+    previousOrderQuery,
+    variables
+  )
+  if (fetchedPreviousOrders.data?.tokenOrders?.length === 0) return null
+  if (assetTimeout === '0') {
+    return fetchedPreviousOrders?.data?.tokenOrders[0]?.tx
+  } else {
+    const expiry = new BigNumber(
+      fetchedPreviousOrders?.data?.tokenOrders[0]?.timestamp
+    ).plus(assetTimeout)
+    const unixTime = new BigNumber(Math.floor(Date.now() / 1000))
+    if (unixTime.isLessThan(expiry)) {
+      return fetchedPreviousOrders?.data?.tokenOrders[0]?.tx
+    } else {
+      return null
+    }
+  }
+}
+
+export async function getAssetPrices(assets: DDO[]): Promise<any> {
+  const priceList: any = {}
+  const poolPriceAssets: string[] = []
+  const poolDTadressDID: any = {}
+  const frePriceAssets: string[] = []
+  const freDTadressDID: any = {}
+  for (const ddo of assets) {
+    if (ddo.price?.type === 'pool') {
```

Comment:
we should stop relying on this from now on.

Reply:
true, started on it after the call pushing it right now, the query's will be a bit bigger because we will search all assets for both fre or poll but as far i see in local testing no big difference

Reply:
you were doing that already, that's why there is no difference